pkg/executor/importer/chunk_process.go

```diff
@@ -418,6 +423,153 @@ func (p *localSortChunkProcessor) deliverLoop(ctx context.Context) error {
 	return nil
 }
 
+func (p *dataDeliver) logFields() []zap.Field {
+	return []zap.Field{
+		zap.Duration("deliverDur", p.deliverTotalDur),
+		zap.Object("checksum", &p.checksum),
+	}
+}
+
+// fileChunkEncoder encode data chunk(either a data file or part of a file).
+type queryChunkEncoder struct {
+	rowCh     chan QueryRow
+	chunkInfo *checkpoints.ChunkCheckpoint
+	logger    *zap.Logger
+	encoder   KVEncoder
+	sendFn    func(ctx context.Context, kvs []deliveredRow) error
+
+	// total duration takes by read/encode/deliver.
+	readTotalDur   time.Duration
+	encodeTotalDur time.Duration
+}
+
+var _ chunkEncoder = (*queryChunkEncoder)(nil)
+
+func (*queryChunkEncoder) init() error {
+	return nil
+}
+
+func (e *queryChunkEncoder) encodeLoop(ctx context.Context) error {
```

Comment:
maybe in future we can unify it with `fileChunkEncoder`? Most logic is shared, the difference is how to read data.

Reply:
https://github.com/pingcap/tidb/pull/50341/commits/6c1c867d973b3a423134c128b9fb16dbbde0632c